.github/workflows/test.yml

```diff
@@ -272,20 +218,10 @@ jobs:
           E2E_KATHY_MESSAGES: '20'
           RUST_BACKTRACE: 'full'
 
-  e2e:
```

Comment:
We don't need this anymore?

Reply:
we originally had to it be backwards compatible when e2e was a single job, pre-matrix. nowadays in the branch protection settings we require the specific cases in the matrix to pass, as opposed to the overall e2e job

also given that we don't want to fail-fast in the e2e-matrix, we'll always hit this e2e job after the matrix completes and this step will always pass regardless of whether any job in the matrix failed. we could change this behaviour, but given our existing branch protection settings I think it's dead weight